### PR TITLE
Number of up crossings construction

### PR DESCRIPTION
**Pull request by alvaro-castillo**

### CODE REVIEW

File: skfda/exploratory/stats/_functional_transformers.py [hunk 1]
```diff
@@ -75,3 +75,112 @@ def local_averages(
...
+            levels: sequence of numbers including the levels
+            we want to consider for the crossings.
+        Returns:
+            ndarray of shape (n_levels, n_samples)\
+            with the values of the counters.
+
+    Example:
+
+    We import the Medflies dataset and for simplicity we use
+    the first 50 samples.
```
Comment (vnmabus) [inline skfda/exploratory/stats/_functional_transformers.py:110]: Use 5 samples, not 50.

File: skfda/exploratory/stats/_functional_transformers.py [hunk 2]
```diff
@@ -75,3 +75,112 @@ def local_averages(
...
+    curves = data.data_matrix
+
+    distances = np.asarray([
+        level - curves
+        for level in levels
+    ])
+
+    points_greater = distances >= 0
+    points_smaller = distances <= 0
+    points_smaller_rotated = np.roll(points_smaller, -1, axis=2)
```
Comment (vnmabus) [inline skfda/exploratory/stats/_functional_transformers.py:181]: Shift, don't roll.

File: skfda/exploratory/stats/_functional_transformers.py [hunk 3]
```diff
@@ -75,3 +75,112 @@ def local_averages(
         for i in range(n_intervals)
     ]
     return np.asarray(integrated_data)
+
+
+def number_up_crossings(
+    data: FDataGrid,
+    levels: np.ndarray,
```
Comment (vnmabus) [inline skfda/exploratory/stats/_functional_transformers.py:82]: Use more specific Numpy types.

File: skfda/exploratory/stats/_functional_transformers.py [hunk 4]
```diff
@@ -75,3 +75,112 @@ def local_averages(
...
+    >>> dataset = fetch_medflies()
+    >>> X = dataset['data'][:50]
+
+    Then we decide the level we want to consider (in our case 40)
+    and call the function with the dataset. The output will be the number of
+    times each curve cross the level 40 growing.
+    >>> from skfda.exploratory.stats import number_up_crossings
+    >>> import numpy as np
+    >>> number_up_crossings(X, np.asarray([40]))
+    array([[[6],
```
Comment (vnmabus) [inline skfda/exploratory/stats/_functional_transformers.py:121]: Shouldn't the returned array have dimensions `n_samples` x `len(levels)`?

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/exploratory/stats/_functional_transformers.py (hunk 4 above)
Comment (alvaro-castillo) [reply skfda/exploratory/stats/_functional_transformers.py:121]: I have prioritized len(levels) over n_samples Therefore shape will be (len(levels), n_samples, n_dimensions). Is it correct, or should I reshape it?

Comment (vnmabus) [reply skfda/exploratory/stats/_functional_transformers.py:121]: In order for the result to be usable in sklearn it should be a 2d array and the first dimension should be `n_samples`.

File: skfda/exploratory/stats/_functional_transformers.py [hunk 5]
```diff
@@ -101,75 +102,30 @@ def number_up_crossings(
...
-            ndarray of shape (n_levels, n_samples)\
+            ndarray of shape (n_samples, len(levels))\
             with the values of the counters.
 
     Example:
 
     We import the Medflies dataset and for simplicity we use
-    the first 50 samples.
+    the first 5 samples.
     >>> from skfda.datasets import fetch_medflies
```
Comment (vnmabus) [inline skfda/exploratory/stats/_functional_transformers.py:112]: I think a example with manually defined curves is probably more illustrative.

**Review by vnmabus — APPROVED** [review]
LGTM

**Pull request merged**